madanalysis/layout/histogram.py

```diff
@@ -1,125 +1,144 @@
+        if self.ymin != [] or self.ymax != []:
+            inform = inform + " " + str(self.ymin) + " " + str(self.ymax)
+        logging.getLogger("MA5").info(inform)
 
         # Data
         self.positive.Print()
         self.negative.Print()
         self.summary.Print()
 
+    def CreateHistogram(self):
+        pass
+
+    def FinalizeReading(self, main, dataset):
 
-    def FinalizeReading(self,main,dataset):
+        self.positive.convert_to_numpy()
+        self.negative.convert_to_numpy()
+        self.summary.convert_to_numpy()
 
         # Statistics
-        self.summary.nevents   = self.positive.nevents   + self.negative.nevents
-        self.summary.nentries  = self.positive.nentries  + self.negative.nentries
+        self.summary.nevents = self.positive.nevents + self.negative.nevents
+        self.summary.nentries = self.positive.nentries + self.negative.nentries
 
         # sumw
-        self.summary.sumw      = self.positive.sumw      - self.negative.sumw
-        if self.summary.sumw<0:
-            self.summary.sumw=0
+        self.summary.sumw = np.clip(self.positive.sumw - self.negative.sumw, 0, None)
 
         # sumw2
-        self.summary.sumw2     = self.positive.sumw2     - self.negative.sumw2
-        if self.summary.sumw2<0:
-            self.summary.sumw2=0
+        self.summary.sumw2 = np.clip(self.positive.sumw2 - self.negative.sumw2, 0, None)
 
         # sumwx
-        self.summary.sumwx     = self.positive.sumwx     - self.negative.sumwx
+        self.summary.sumwx = self.positive.sumwx - self.negative.sumwx
         # no correction on it
 
         # sumw2x
-        self.summary.sumw2x    = self.positive.sumw2x    - self.negative.sumw2x
+        self.summary.sumw2x = self.positive.sumw2x - self.negative.sumw2x
         # no correction on it
 
         # underflow
-        self.summary.underflow = self.positive.underflow - self.negative.underflow
-        if self.summary.underflow<0:
-            self.summary.underflow=0
+        self.summary.underflow = np.clip(
+            self.positive.underflow - self.negative.underflow, 0, None
+        )
 
         # overflow
-        self.summary.overflow  = self.positive.overflow  - self.negative.overflow
-        if self.summary.overflow<0:
-            self.summary.overflow=0
+        self.summary.overflow = np.clip(
+            self.positive.overflow - self.negative.overflow, 0, None
+        )
+
+        # compute mean and uncertainties for the statistics
+        # ! @jackaraz: this portion of the code should be changed to accomodate different types of
+        # ! PDF + scale unc combination for now its just mean and std
+        for tp in [
+            "nevents",
+            "nentries",
+            "sumw",
+            "sumw2",
+            "sumwx",
+            "sumw2x",
+            "underflow",
+            "overflow",
+        ]:
+            # compute unc shape: (lower, upper)
```

Comment:
we need to discuss this. It looks weird.

Reply:
This comment is not explicit I don't understand what you are pointing at

Reply:
I don't understand your comment about mean and stddev.

Reply:
that is old comment, I already moved on from plotting mean and std, I should clean those parts